azurelinuxagent/common/event.py

```diff
@@ -429,10 +457,30 @@ def add_event(name, op=WALAEventOperation.Unknown, is_success=True, duration=0,
 
 
 def add_log_event(level, message, reporter=__event_logger__):
+    """
+    # See #1035 for not adding warning/error statements here (any where in the call-tree).
+    The issue with #1035 was never-ending machinery generating events continuously and stretching the system.
+
+    At t=0 - Assume 999 events are already in the events folder.
+    At t=1 - Now, assuming that a new log.WARN [1] event comes in and needs to be written - when saving this event on
+    the disk `save_event` would generate another log.WARN [2] as it warns that there are too many events on the dist.
```

Comment:
I have a question, our normal logger just writes to the waagent.log file right? It doesnt create a new file to be saved in the events directory? Only the telemetry events are saved there. So if the `save_event` function writes a warn `logger.warn` how/why would that write to our events directory to cause this recursion?

Reply:
The PR adds a telemetry appender to the logger, so now log.warn does produce a telemetry event. When creating the event (e.g. while saving to disk) somebody does log.warn again, that will try to produce a telemetry event which...

Reply:
Yes got it, @vrdmr explained it to me offline. Thanks for the clarification guys! :)